support/collect/lazy_collection.go

```diff
@@ -0,0 +1,871 @@
+		output := make(chan T)
+		go func() {
+			defer close(output)
+			seen := make(map[string]bool)
+			for item := range input {
+				key := fmt.Sprintf("%v", item)
+				if !seen[key] {
+					seen[key] = true
+					output <- item
+				}
+			}
+		}()
+		return output
+	})
+
+	return &LazyCollection[T]{
+		generator: lc.generator,
+		pipeline:  newPipeline,
+	}
+}
+
+func (lc *LazyCollection[T]) UniqueBy(keyFunc func(T) string) *LazyCollection[T] {
+	newPipeline := make([]func(<-chan T) <-chan T, len(lc.pipeline))
+	copy(newPipeline, lc.pipeline)
+
+	newPipeline = append(newPipeline, func(input <-chan T) <-chan T {
+		output := make(chan T)
+		go func() {
+			defer close(output)
+			seen := make(map[string]bool)
+			for item := range input {
+				key := keyFunc(item)
+				if !seen[key] {
+					seen[key] = true
+					output <- item
+				}
+			}
+		}()
+		return output
+	})
+
+	return &LazyCollection[T]{
+		generator: lc.generator,
+		pipeline:  newPipeline,
+	}
+}
+
+func (lc *LazyCollection[T]) When(condition bool, fn func(*LazyCollection[T]) *LazyCollection[T]) *LazyCollection[T] {
+	if condition {
+		return fn(lc)
+	}
+	return lc
+}
+
+func (lc *LazyCollection[T]) Where(params ...interface{}) *LazyCollection[T] {
+	switch len(params) {
+	case 1:
+		// where(callback)
+		if callback, ok := params[0].(func(T) bool); ok {
+			return lc.Filter(func(item T, _ int) bool {
+				return callback(item)
+			})
+		}
+		return lc
+	case 2:
+		// where(field, value) - assumes '=' operator
+		if field, ok := params[0].(string); ok {
+			return lc.Filter(func(item T, _ int) bool {
+				return compareFieldValue(item, field, "=", params[1])
+			})
+		}
+		return lc
+	case 3:
+		// where(field, operator, value)
+		if field, ok := params[0].(string); ok {
+			if operator, ok := params[1].(string); ok {
+				return lc.Filter(func(item T, _ int) bool {
+					return compareFieldValue(item, field, operator, params[2])
+				})
+			}
+		}
+		return lc
+	default:
+		return lc
+	}
+}
+
+func (lc *LazyCollection[T]) WhereIn(field string, values []interface{}) *LazyCollection[T] {
+	valueMap := make(map[string]bool)
+	for _, v := range values {
+		valueMap[fmt.Sprintf("%v", v)] = true
+	}
+
+	return lc.Filter(func(item T, _ int) bool {
+		fieldValue := getFieldValue(item, field)
+		if fieldValue == nil {
+			return false
+		}
+		return valueMap[fmt.Sprintf("%v", *fieldValue)]
+	})
+}
+
+func (lc *LazyCollection[T]) WhereNotIn(field string, values []interface{}) *LazyCollection[T] {
+	valueMap := make(map[string]bool)
+	for _, v := range values {
+		valueMap[fmt.Sprintf("%v", v)] = true
+	}
+
+	return lc.Filter(func(item T, _ int) bool {
+		fieldValue := getFieldValue(item, field)
+		if fieldValue == nil {
+			return true
+		}
+		return !valueMap[fmt.Sprintf("%v", *fieldValue)]
+	})
+}
+
+func (lc *LazyCollection[T]) Zip(other *LazyCollection[T]) [][]T {
+	var result [][]T
+	ch1 := lc.execute()
+	ch2 := other.execute()
+
+	for item1 := range ch1 {
+		item2, ok2 := <-ch2
```

Comment:
The same as above, `ch1` and `ch2` may be leak.


```suggestion
	ch1 := lc.execute()
	ch2 := other.execute()
	
	defer func() {
        go func() { for range ch1 {} }()
        go func() { for range ch2 {} }()
    }()

	for item1 := range ch1 {
		item2, ok2 := <-ch2
```